addons/test_mail/tests/test_mail_multicompany.py

```diff
@@ -253,62 +255,136 @@ def test_post_wo_access(self):
-        test_activity = next(
-            a for a in data["Store"]["activityGroups"]
-            if a['model'] == 'mail.test.multi.company.with.activity'
-        )
-        self.assertEqual(
-            test_activity,
-            {
-                "icon": "/base/static/description/icon.png",
-                "id": self.env["ir.model"]._get_id("mail.test.multi.company.with.activity"),
-                "model": "mail.test.multi.company.with.activity",
-                "name": "Test Multi Company Mail With Activity",
-                "overdue_count": 0,
-                "planned_count": 0,
-                "today_count": 2,
-                "total_count": 2,
-                "type": "activity",
-                "view_type": "list",
-            }
-        )
-        params = {
-            "systray_get_activities": True,
-            "context": {"allowed_company_ids": [self.company_2.id]},
-        }
-        data = self.make_jsonrpc_request("/mail/data", params)
-        test_activity = next(
-            a for a in data["Store"]["activityGroups"]
-            if a['model'] == 'mail.test.multi.company.with.activity'
-        )
-        self.assertEqual(
-            test_activity,
-            {
-                "icon": "/base/static/description/icon.png",
-                "id": self.env["ir.model"]._get_id("mail.test.multi.company.with.activity"),
-                "model": "mail.test.multi.company.with.activity",
-                "name": "Test Multi Company Mail With Activity",
-                "overdue_count": 0,
-                "planned_count": 0,
-                "today_count": 1,
-                "total_count": 1,
-                "type": "activity",
-                "view_type": "list",
-            }
-        )
+        test_records, test_records_admin = test_records_all[:2], test_records_all[2:]
+        # Schedule an employee and admin todo activity for each records
+        employee_act_by_record = dict()
+        for record in test_records_all:
+            record.activity_schedule("test_mail.mail_act_test_todo", user_id=user_admin.id)
+        for record in test_records_all:
+            employee_act_by_record[record] = record.activity_schedule(
+                "test_mail.mail_act_test_todo", user_id=user_employee.id)
+        other_activity_group_model = 'mail.activity'
+
+        self.assertTrue((company_1_all | company_2_admin_only) <= user_admin.company_ids)
+        self.assertEqual(company_1_all, user_employee.company_ids)
+        # We test the outcome of systray_get_activities for different couple of user and allowed companies
+        for description, (user, allowed_company_ids), (expected_other_activities, expected_n_std_activities) in (
+                (
+                        "Admin see only activities of records of allowed companies (company_1).",
+                        (user_admin, company_1_all.ids),
+                        (False, 1 + 2),
```

Comment:
Can't we give (and compare) expected activities instead, to be clearer ? Otherwise you might end up with "2 different from 3" but that does not help you understanding what went wrong